pages/wiki/openembedded.md

```diff
@@ -3,10 +3,243 @@ title: OpenEmbedded
+# OpenEmbedded
+
+[OpenEmbedded](http://www.openembedded.org/wiki/Main_Page) (OE) is the build system used by AsteroidOS. It provides a way to easily maintain embedded Linux distributions. OE is also used by projects such as WebOS-Ports, SHR or Ångström. OpenEmbedded is supported by the [Yocto project](https://www.yoctoproject.org/) from the Linux Foundation and provides a rock-solid basis for AsteroidOS.
+
+OpenEmbedded uses the [`bitbake`](https://docs.yoctoproject.org/bitbake/) tool which is developed in python to generate images (like asteroid-image) from packages (like asteroid-launcher-dev) built from recipes (ex: asteroid-launcher_git.bb) provided in layers (ex: meta-asteroid).
+
+Layers are git repositories containing a bunch of related recipes, example of layers include board support packages (BSPs) like [meta-dory-hybris](https://github.com/AsteroidOS/meta-smartwatch/tree/master/meta-dory) or meta-rockchip which describe a way to support a new machine, but also UI layers like meta-xfce or meta-gnome which describe the building process of graphic components.
+
+A recipe describes how to fetch, patch, configure, compile, install, package (in .rpm, .deb or .ipk) and test a piece of software. Bitbake handles all those operations and dependencies between them and between recipes. In the end it can generate images or an SDK for multiple targets.
+
+Contributing to the AsteroidOS’s OE architecture mostly consists in maintaining the [meta-asteroid](https://github.com/AsteroidOS/meta-asteroid) and [meta-smartwatch](https://github.com/AsteroidOS/meta-smartwatch) repositories which contain the recipes related to AsteroidOS and the watches' BSPs.
+In depth info can be found in the [Yocto Project's manual](https://docs.yoctoproject.org/index.html)
+
 <img src="{{assets}}/img/openembedded.png" style="width:100%"/>
+
+
+# A Quick Tutorial
+
+---
+
+The rest of this description assumes that you have already read and followed the [Building AsteroidOS]({{rel 'wiki/building-asteroidos'}}) instructions, and that you have built AsteroidOS for at least one platform (a particular watch or the emulator), and that you are familiar with [`git`](https://www.atlassian.com/git/tutorials) and the [Linux](https://www.linux.org/forums/linux-beginner-tutorials.123/) command line.
+
+This tutorial will introduce you to the following topics:
+
+ - [Changing an existing application:](#changing-an-existing-application) You don't like how something behaves or want to fix an issue in an existing application? This will introduce you to the tooling used to allow you to make these changes.
+ - [Adding a new application:](#adding-a-new-application) Use the tooling to create a new application recipe and test it on a watch.
+ - [Sharing your changes:](#sharing-your-changes) You've made changes on your local build of AsteroidOS. How do you share these changes? This will be explained in this part of the tutorial.
+
+
+# Changing an Existing Application
```

Comment:
These are logically "under" "Quick Tutorials" so the titles should be smaller so the hierarchy stays consistent

Reply:
Fixed.